hydra-cardano-api/src/Hydra/Cardano/Api/BlockHeader.hs

```diff
@@ -30,9 +19,26 @@ genBlockHeader = do
 -- certain slot.
 genBlockHeaderAt :: SlotNo -> Gen BlockHeader
 genBlockHeaderAt slotNo = do
-  headerHash <- unsafeBlockHeaderHashFromBytes . BS.pack <$> vectorOf 32 arbitrary
+  headerHash <- genBlockHeaderHash
   blockNo <- BlockNo <$> arbitrary
   pure $ BlockHeader slotNo headerHash blockNo
 
+-- | Generate a random block header hash.
+genBlockHeaderHash :: Gen (Hash BlockHeader)
+genBlockHeaderHash =
+  unsafeBlockHeaderHashFromBytes . BS.pack <$> vectorOf 32 arbitrary
+ where
+  unsafeBlockHeaderHashFromBytes bytes =
+    case deserialiseFromRawBytes (proxyToAsType Proxy) bytes of
+      Left e ->
+        error $
```

Comment:
Out of interest, did you add this because this error actually happens? Why does it happen?

Reply:
I have not changed the semantics, I just moved this function into a local `where` block as it was not used anywhere else.

Reply:
Trying to answer your question: This is a "should not happen" error as the Proxy defines on compile time how many bytes are required. Unless the `BlockHeader` hash instance is not changed, the `32` generated bytes will always suffice.

We _could_ make this more robust by using the underlying type classes (also used by the `deserialiseFromRawBytes` implementation) to generate always the right amount of bytes. This would require more importants but could avoid the hard-coded `32`. WDYT?

Reply:
Ah very true; sorry, I somehow missed that that error stuff was already there.

Reply:
> This would require more imports but could avoid the hard-coded 32. WDYT?

More imports and less hard-coded values (and no error handling!) seems like a fairly good trade-off to me!

Reply:
I was wrong. They don't even use something specific to `Hash BlockHeader` type and hard-code a `32` too: https://github.com/input-output-hk/cardano-api/blob/a7af2f29dbc7ae923a7140dddbbbb0f89f760046/cardano-api/internal/Cardano/Api/Block.hs#L241